source/common/conn_pool/conn_pool_base.cc

```diff
@@ -443,8 +447,8 @@ void ConnPoolImplBase::onConnectionEvent(ActiveClient& client, absl::string_view
     client.releaseResources();
 
     dispatcher_.deferredDelete(client.removeFromList(owningList(client.state())));
-    if (incomplete_stream) {
-      checkForDrained();
+    if (incomplete_stream || !is_draining_) {
```

Comment:
Why do we check for !is_draining_ here now? Add a comment?

Reply:
I'm not sure. That came in from the code in the previous PR. I'll try to figure it out and add a comment or revert.

Reply:
Commented